In [9]:
import pandas as pd

**Task_1**

Оскільки ймовірність того, що при перевірці деталь виявиться стандартною дорівнює 0.6, то відповідно ймовірність того, що деталь виявиться бракованою дорівнюватиме 0.4.
Розглянемо варіанти ймовірностей, що можуть бути в результаті перевірки 3-ох деталей:
1. Всі деталі стандартні P = 0.6 * 0.6 * 0.6 = 0.216
2. Всі деталі браковані P = 0.4 + 0.4 + 0.4 = 0.064
3. Дві деталі браковані і одна стандартна P = 0.4 * 0.4 * 0.6 = 0.096 * 3 (варіації) = 0,288
4. Одна деталь бракована і 2 стандартні P = 0.4 * 0.6 * 0.6 = 0.144 * 3 (варіації) = 0,432

**Закон розподілу системи двох дискретних випадкових величин X - появи числа бракованих деталей і Y - появи числа стандартних деталей:**

In [163]:
tabledata = [["", "", "", "", "", ""],
             ["", 0, 0, 0, 0.064, 0.064],
             ["", 0, 0, 0.288, 0, 0.288],
             ["", 0, 0.432, 0, 0, 0.432],
             ["", 0.216, 0, 0, 0, 0.216],
             ["", 0.216, 0.432, 0.288, 0.064, 1]]

In [164]:
pd.DataFrame(tabledata, 
             index=["Y", 0, 1, 2, 3, "P(x)"],
             columns=["X", 0, 1, 2, 3, "P(y)"])


X      0      1      2      3   P(y)
Y                                        
0           0      0      0  0.064  0.064
1           0      0  0.288      0  0.288
2           0  0.432      0      0  0.432
3       0.216      0      0      0  0.216
P(x)    0.216  0.432  0.288  0.064      1

Для обчислення коефіцієнту кореляції $r_{xy}$ необхідно знайти математичне сподівання, дисперсію, середнє квадратичне відхилення величин X та Y, а також коваріацію cov(X,Y).

**Знайдемо вищезазначені величини:**

M(X) = 0 * 0.216 + 1 * 0.432 + 2 * 0.288 + 3 * 0.064 = 1.2

In [170]:
0 * 0.216 + 1 * 0.432 + 2 * 0.288 + 3 * 0.064

1.2

$M(X^2) = 0^2 * 0.216 + 1^2 * 0.432 + 2^2 * 0.288 + 3^2 * 0.064 = 2.16$ 

In [169]:
0 ** 2 * 0.216 + 1 ** 2 * 0.432 + 2 ** 2 * 0.288 + 3 ** 2 * 0.064

2.16

$D(X) = M(X^2) - (M(X))^2 = 2.16 - 1.2^2 = 2.16 - 1.44 = 0.72$

In [171]:
2.16 - 1.2 ** 2

0.7200000000000002

$σ_x = \sqrt {D(X)} = \sqrt0.72 = 0.85$


In [172]:
0.72 ** 0.5

0.848528137423857

M(Y) = 0 * 0.064 + 1 * 0.288 + 2 * 0.432 + 3 * 0.216 = 1.8

In [173]:
0 * 0.064 + 1 * 0.288 + 2 * 0.432 + 3 * 0.216

1.7999999999999998

$M(Y^2) = 0^2 * 0.064 + 1^2 * 0.288 + 2^2 * 0.432 + 3^2 * 0.216 = 3.96$ 

In [180]:
0 ** 2 * 0.064 + 1 ** 2 * 0.288 + 2 ** 2 * 0.432 + 3 ** 2 * 0.216

3.96

$D(Y) = M(Y^2) - (M(Y))^2 = 3.96 - 3.24 = 0.72$

In [181]:
3.96 - 1.8 ** 2

0.7199999999999998

$σ_y = \sqrt {D(Y)} = \sqrt0.72 = 0.85$

In [174]:
0.72 ** 0.5

0.848528137423857

M(XY) = 2 * 1 * 0.288 + 1 * 2 * 0.432 = 1.44

In [175]:
2 * 1 * 0.288 + 1 * 2 * 0.432

1.44

cov(X,Y) = M(XY) - M(X)M(Y) = 1.44 - 1.2 * 1.8 = -0.72

In [182]:
1.44 - 1.2 * 1.8

-0.7200000000000002

**Знайдемо коефіцієнт кореляції:**

$r_{xy} = cov(X,Y) / σ_x σ_y$ = -0.72 / (0.85 * 0.85) = -1.00

In [179]:
-0.72 / (0.85 * 0.85)

-0.9965397923875433

**Task_2**

*2.1 Обчислити очікувану норму прибутку по кожній з акцій*

M(X) = 10 * 0.1 + 5 * 0.3 + 2 * 0.2 + 1 * 0.3 + (-5 * 0.1) = 2.7

In [185]:
10 * 0.1 + 5 * 0.3 + 2 * 0.2 + 1 * 0.3 + (-5 * 0.1) 

2.6999999999999997

M(Y) = 20 * 0.1 + 10 * 0.3 + 2 * 0.2 + (-2 * 0.3) + (-8 * 0.1) = 4

In [186]:
20 * 0.1 + 10 * 0.3 + 2 * 0.2 + (-2 * 0.3) + (-8 * 0.1)

4.000000000000001

*2.2 оцінити степінь ризику обох акцій за допомогою середньоквадратичного відхилення*

$M(X^2) = 10^2 * 0.1 + 5^2 * 0.3 + 2^2 * 0.2 + 1^2 * 0.3 + (-5^2 * 0.1) = 21.1$ 

In [190]:
10 ** 2 * 0.1 + 5 ** 2 * 0.3 + 2 ** 2 * 0.2 + 1 ** 2 * 0.3 + ((-5) ** 2 * 0.1)

21.1

$D(X) = M(X^2) - (M(X))^2 = 21.1 - 2.7^2 = 21.7 - 7.29 = 14.41$

$σ_x = \sqrt {D(X)} = \sqrt14.41 = 3.80$

In [193]:
14.41 ** 0.5

3.7960505792204615

$M(Y^2) = 20^2 * 0.1 + 10^2 * 0.3 + 2^2 * 0.2 + (-2^2 * 0.3) + (-8^2 * 0.1) = 78.4$ 

In [196]:
20 ** 2 * 0.1 + 10 ** 2 * 0.3 + 2 ** 2 * 0.2 + ((-2) ** 2 * 0.3) + ((-8) ** 2 *0.1)

78.4

$D(Y) = M(Y^2) - (M(Y))^2 = 78.4 - 16 = $

In [197]:
78.4 - 4 ** 2

62.400000000000006

$σ_y = \sqrt {D(Y)} = \sqrt62.4 = 7.90$

In [198]:
62.4 ** 0.5

7.899367063252599

**Оскільки значення середнього квадратичного відхилення по акції Х ($σ_x = 3.80$) є меншим за значення середнього квадратичного відхилення по акції Y ($σ_y = 7.90$), то ступінь ризику акцій X є меншим.**

*2.3 проаналізувати наявність зв’язку між акціями X та Y за допомогою коефіцієнтів коваріації та кореляції*

M(XY) = 20 * 10 * 0.01 + 20 * 5 * 0.03 + 20 * 2 * 0.02 + 20 * 1 * 0.03 + (20 * (-5) * 0.01) + 10 * 10 * 0.03 + 10 * 5 * 0.09 + 10 * 2 * 0.06 + 10 * 1 * 0.09 + (10 * (-5) * 0.03) + 2 * 10 * 0.02 + 2 * 5 * 0.06 + 2 * 2 * 0.04 + 2 * 1 * 0.06 + (2 * (-5) * 0.02) + ((-2) * 10 * 0.03) + ((-2) * 5 * 0.09) + ((-2) * 2 * 0.06) + ((-2) * 1 * 0.09) + ((-2) * (-5) * 0.03) + ((-8) * 10 * 0.01) + ((-8) * 5  * 0.03) + ((-8) * 2 * 0.02) + ((-8) * 1 * 0.03) + ((-8) * (-5) * 0.01) = 10.80

In [200]:
20 * 10 * 0.01 + 20 * 5 * 0.03 + 20 * 2 * 0.02 + 20 * 1 * 0.03 + (20 * (-5) * 0.01) + 10 * 10 * 0.03 + 10 * 5 * 0.09 + 10 * 2 * 0.06 + 10 * 1 * 0.09 + (10 * (-5) * 0.03) + 2 * 10 * 0.02 + 2 * 5 * 0.06 + 2 * 2 * 0.04 + 2 * 1 * 0.06 + (2 * (-5) * 0.02) + ((-2) * 10 * 0.03) + ((-2) * 5 * 0.09) + ((-2) * 2 * 0.06) + ((-2) * 1 * 0.09) + ((-2) * (-5) * 0.03) + ((-8) * 10 * 0.01) + ((-8) * 5  * 0.03) + ((-8) * 2 * 0.02) + ((-8) * 1 * 0.03) + ((-8) * (-5) * 0.01)

10.799999999999999

cov(X,Y) = M(XY) - M(X)M(Y) = 10.8 - 2.7 * 4 = 0

In [201]:
10.8 - 2.7 * 4

0.0

$r_{xy} = cov(X,Y) / σ_x σ_y$ = 0 / (3.8 * 7.90) = 0

**Оскільки коваріація рівна нулю, то величини X та Y будуть некорельованими, тобто не залежними одна від одної.**